libsurfman/src/surface.c

```diff
@@ -213,13 +285,26 @@ void surfman_surface_update_pfn_arr (surfman_surface_t * surface,
                                 const xen_pfn_t * pfns)
 {
   struct surface_priv *p = PRIV(surface);
+  xc_dominfo_t info;
+  xen_pfn_t *_pfns;
+  size_t i, j;
 
   pthread_mutex_lock (&p->lock);
   p->type = TYPE_PFN_ARR;
   p->u.pfn_arr.domid = surface->pages_domid;
   p->u.pfn_arr.arr = pfns;
   if (p->baseptr)
     update_mapping (p, surface->page_count * XC_PAGE_SIZE);
+
+  if (xc_domid_getinfo (surface->pages_domid, &info) != 1)
+      surfman_error ("Cannot retrieve dom%u information.", surface->pages_domid);
+  else if (info.hvm)
+    {
+      if (surface_pin_cacheattr (surface, XC_MAP_CACHEATTR_WC))
+        surfman_error ("Failed to pin dom%u surface cache-attributes.", surface->pages_domid);
```

Comment:
Would the error code returned on failure here be helpful to print for assisting diagnosis?

Reply:
I presumed it would not, but reading through `hvm_set_mem_pinned_cacheattr`:
- EOPNOTSUPP: for non-hvm domain
- EINVAL: for invalid ranges or types
- ENOMEM: allocation failure
- EBUSY: changing attributes on a sub-range of a range already pinned.

So yes we should log that at least.